### PR TITLE
refs[TASK][31123]: Create request get data

### PR DESCRIPTION
**Pull request by Domhoan**
## Related Tickets
- [#TASK 31123 : Create request get data](https://edu-redmine.sun-asterisk.vn/issues/31123)
## WHAT
- Create request get data: Match, Stat, Team
## HOW
- Make internet connection & get data
## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sun/premierleague/data/source/remote/api/ApiService.kt [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                connect()
+            }
+            val respondCode = urlConnection.responseCode
+            if (respondCode == HttpURLConnection.HTTP_OK) {
+                url.openStream().bufferedReader().use(BufferedReader::readText)
+            } else throw IOException(urlConnection.responseMessage)
+        } finally {
+            urlConnection?.disconnect()
+        }
+    }
```
Comment (huytq-2422) [inline app/src/main/java/com/sun/premierleague/data/source/remote/api/ApiService.kt]: việc lấy dữ liệu ông phải thực hiện đồng bộ, sử dụng asyc task trong trường hợp này nhé

File: app/src/main/java/com/sun/premierleague/data/source/remote/api/ApiService.kt [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import com.sun.premierleague.data.model.Time
+import org.json.JSONArray
+import java.io.BufferedReader
+import java.io.IOException
+import java.net.HttpURLConnection
+import java.net.URL
+
+object ApiService {
+
+    private fun queryMatch(time: Time): List<MatchItem> {
```
Comment (NguyenTaTrung) [inline app/src/main/java/com/sun/premierleague/data/source/remote/api/ApiService.kt]: hàm này để private hả chú?

Comment (Domhoan) [reply app/src/main/java/com/sun/premierleague/data/source/remote/api/ApiService.kt]: ok Trung

File: gradle.properties [hunk 3]
```diff
@@ -19,3 +19,4 @@ android.useAndroidX=true
 android.enableJetifier=true
 # Kotlin code style for this project: "official" or "obsolete":
 kotlin.code.style=official
+API_KEY=20fd331cca48d4dd9fa2d05f5a5b31d0071361963007255bef964c3e89e009ea
```
Comment (viet06061999) [inline gradle.properties]: Hide API_KEY đi anh.
https://viblo.asia/p/cac-thu-thuat-de-giau-secret-keypassword-khoi-source-control-va-ung-dung-voi-ci-trong-lap-trinh-android-PwRGgmOnkEd

**Review by NguyenTaTrung — APPROVED** [review]

File: app/src/main/java/com/sun/premierleague/data/source/remote/RemoteAsyncTask.kt [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        try {
+            handle(params[0])
+                ?: throw Exception(R.string.err_fist_param_not_be_null.toString())
+        } catch (e: Exception) {
+            exception = e
+            null
+        }
+
+    override fun onPostExecute(result: T?) {
+        result?.let(callback::onSuccess) ?: (callback::onFailure)
```
Comment (datdt-1935) [inline app/src/main/java/com/sun/premierleague/data/source/remote/RemoteAsyncTask.kt]: onFailure cái exception ra chứ em?

File: app/src/main/java/com/sun/premierleague/data/source/remote/api/ApiService.kt [hunk 5]
```diff
@@ -0,0 +1,37 @@
+package com.sun.premierleague.data.source.remote.api
+
+import android.net.Uri
+import com.sun.premierleague.BuildConfig
+import com.sun.premierleague.data.model.Time
+
```
Comment (tuannm-1276) [inline app/src/main/java/com/sun/premierleague/data/source/remote/api/ApiService.kt:6]: blank

**Pull request merged**